common/djangoapps/util/keyword_substitution.py

```diff
@@ -53,30 +46,36 @@ def substitute_keywords(string, user=None, course=None):
     the loop below, and reduces the need for piling up if elif else statements
     when the keyword pool grows.
     """
-    if user is None or course is None:
-        # Cannot proceed without course and user information
-        return string
 
-    for key, func in KEYWORD_FUNCTION_MAP.iteritems():
+    # do this lazily to avoid unneeded database hits
+    KEYWORD_FUNCTION_MAP = {
```

Comment:
Yeah, it looked like the map was hard coded anyway, so I didn't think it would be a big deal to move it here. Plus it's common, so both lms and studio can use it.